### PR TITLE
League of Moveable Type fonts: update

### PR DESCRIPTION
**Pull request by minijackson**
###### Description of changes

Related: #204925

For fonts that already were a separate package, migrate them to `stdenvNoCC` (orbitron, raleway). Add the rest as separate package.

Currently, fonts are unavailable from their website, I think they moved them to GitHub, which means today those fonts cannot be rebuilt without `tarballs.nixos.org`.

The league-of-moveable-type collection was change to be a `symlinkJoin`.

Pinging @bergey and @Profpatsch as maintainers of the collection package: I didn't know if you wanted to be maintainer of each individual package so I took the conservative approach.

@Profpatsch I also migrated the Raleway font from the `impallari` repo to the `theleagueof` one, since it is the most recent one. I think they just added a variable font.

Pinging @leenaars as maintainer of Orbitron, I changed the date and the date format for coherence, if that's okay with you.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - They show up under `gnome-font-viewer`
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Eisfunke** [comment]
I've noticed that these fonts are out-of-date as well, great to see somebody already did the work here, thanks!

> @Profpatsch I also migrated the Raleway font from the `impallari` repo to the `theleagueof` one, since it is the most recent one. I think they just added a variable font.

Not that it matters too much, but as a note: they've also integrated improvements on the Impallari version from even later forks besides the variable variant (see e.g. https://github.com/theleagueof/raleway/issues/2).

File: pkgs/data/fonts/league-script-number-one/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  installPhase = ''
+    runHook preInstall
+
+    install -D -m444 -t $out/share/fonts/opentype $src/*.otf
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "A modern, coquettish script font that sits somewhere between your high school girlfriend’s love notes and handwritten letters from the '20s.";
```
Comment (NickCao) [inline pkgs/data/fonts/league-script-number-one/default.nix:23]: No trailing dot in descriptions, also keep the sentence short and clear, see [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) for guidelines.

File: pkgs/data/fonts/league-script-number-one/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
```
Comment (NickCao) [inline pkgs/data/fonts/league-script-number-one/default.nix:24]: We do not normally keep empty lines between attributes in meta.

**Review by NickCao — COMMENTED** [review]
Apart from few formatting issues, this looks perfect, thanks for your great work!

**Comment by bergey** [comment]
Thanks for the tag. As you can see, I have been maintainer in name only. Probably best therefore not to add me to any new packages. Thank you for this excellent work!

**Review by NickCao — APPROVED** [review]

**Pull request merged**